File: samples/server/petstore/java-helidon-server/v4/se/src/main/java/org/openapitools/server/api/StoreService.java
package org.openapitools.server.api;

import java.util.HexFormat;
import java.util.Map;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.openapitools.server.model.Order;
import io.helidon.http.Status;

import io.helidon.webserver.http.HttpRules;
import io.helidon.webserver.http.ServerRequest;
import io.helidon.webserver.http.ServerResponse;
import io.helidon.webserver.http.HttpService;

@io.helidon.common.Generated(value = "org.openapitools.codegen.languages.JavaHelidonServerCodegen",
                             trigger = "tag = 'Store'",
                             version = "stable")
public interface StoreService extends HttpService {

    /**
     * A service registers itself by updating the routing rules.
     * @param rules the routing rules.
     */
    @Override
    default void routing(HttpRules rules) {
        rules.delete("/order/{order_id}", this::deleteOrder);
        rules.get("/inventory", this::getInventory);
        rules.get("/order/{order_id}", this::getOrderById);
        rules.post("/order", this::placeOrder);
    }


    /**
     * DELETE /store/order/{order_id} : Delete purchase order by ID.
     *
     * @param request the server request
     * @param response the server response
     */
    void deleteOrder(ServerRequest request, ServerResponse response);
    /**
     * GET /store/inventory : Returns pet inventories by status.
     *
     * @param request the server request
     * @param response the server response
     */
    void getInventory(ServerRequest request, ServerResponse response);
    /**
     * GET /store/order/{order_id} : Find purchase order by ID.
     *
     * @param request the server request
     * @param response the server response
     */
    void getOrderById(ServerRequest request, ServerResponse response);
    /**
     * POST /store/order : Place an order for a pet.
     *
     * @param request the server request
     * @param response the server response
     */
    void placeOrder(ServerRequest request, ServerResponse response);
}

File: samples/server/petstore/typescript-nestjs-server/src/handlers/StoreService.ts
import { Observable } from 'rxjs';
import { Injectable } from '@nestjs/common';
import { StoreApi } from '../../builds/default/api';
import { Order } from '../../builds/default/models';

@Injectable()
export class StoreService implements StoreApi {

  deleteOrder(
    orderId: string,
    request: Request,
  ): void | Promise<void> | Observable<void> {
    throw new Error('Method not implemented.');
  }

  getInventory(request: Request):
    | { [p: string]: number }
    | Promise<{ [p: string]: number }>
    | Observable<{
        [p: string]: number;
      }> {
    throw new Error('Method not implemented.');
  }

  getOrderById(
    orderId: number,
    request: Request,
  ): Order | Promise<Order> | Observable<Order> {
    throw new Error('Method not implemented.');
  }

  placeOrder(
    order: Order,
    request: Request,
  ): Order | Promise<Order> | Observable<Order> {
    throw new Error('Method not implemented.');
  }
}
